### PR TITLE
Remove lazy behavior from MelScale

### PR DESCRIPTION
**Pull request by hwangjeff**
Rebases #1571; addresses #1569:

"In 0.9.0 we are deprecating the lazy behavior of MelScale because it can make an invalid TorchScript object and it does not align with the design of torchaudio. Now in master branch, we can remove the implementation."

### CODE REVIEW

File: test/torchaudio_unittest/transforms/batch_consistency_test.py [hunk 1]
```diff
@@ -33,13 +33,14 @@ def test_batch_Resample(self):
         self.assertEqual(computed, expected)
 
     def test_batch_MelScale(self):
-        specgram = torch.randn(2, 31, 2786)
+        n_stft = 31
+        specgram = torch.randn(2, n_stft, 2786)
```
Comment (mthrok) [inline test/torchaudio_unittest/transforms/batch_consistency_test.py:37]: Can you use the default value of `n_stft=201`, so that we can catch an error for default value first if that happens in the future?
Also not sure where the 2786 is coming from, but I think we can use something shorter, like 256.

File: test/torchaudio_unittest/transforms/batch_consistency_test.py [hunk 2]
```diff
@@ -33,13 +33,14 @@ def test_batch_Resample(self):
...
 
         # Single then transform then batch
-        expected = torchaudio.transforms.MelScale()(specgram).repeat(3, 1, 1, 1)
+        expected = torchaudio.transforms.MelScale(n_stft=n_stft)(specgram).repeat(3, 1, 1, 1)
```
Comment (mthrok) [inline test/torchaudio_unittest/transforms/batch_consistency_test.py:40]: Note: (follow-up, not the scope of this PR) For batch consistency test, use of repeated does not catch some edge case. https://github.com/pytorch/audio/pull/1315
We should update the logic at some point.

File: test/torchaudio_unittest/transforms/transforms_test.py [hunk 3]
```diff
@@ -56,7 +56,7 @@ def test_AmplitudeToDB(self):
 
     def test_melscale_load_save(self):
         specgram = torch.ones(1, 1000, 100)
-        melscale_transform = transforms.MelScale()
+        melscale_transform = transforms.MelScale(n_stft=1000)
```
Comment (mthrok) [inline test/torchaudio_unittest/transforms/transforms_test.py:59]: Instead of setting `n_stft`, can we change the shape of `specgram` to `1, 201, 100`?
This is more of a personal preference but `1000` is somehow conspicuous.

**Review by mthrok — APPROVED** [review]
Looks good. Left minor comments on test improvements.

I know these are not directly related to the PR, but otherwise we wouldn't be improving the tests.
